lib/ex_aws/sqs.ex

```diff
@@ -495,15 +495,34 @@ defmodule ExAws.SQS do
 
   defp generate_query(action, params) do
     action_string = action |> Atom.to_string() |> Macro.camelize()
+    parser = fetch_parser!()
 
     %ExAws.Operation.Query{
       params: params |> Map.put("Action", action_string),
       service: :sqs,
       action: action,
-      parser: &ExAws.SQS.Parsers.parse/2
+      parser: &parser.parse/2
     }
   end
 
+  defp fetch_parser!() do
+    case Application.fetch_env(:ex_aws_sqs, :parser) do
+      {:ok, parser} ->
+        parser
+
+      :error ->
+        parser =
+          cond do
+            Code.ensure_loaded?(ExAws.SQS.SaxyParser) -> ExAws.SQS.SaxyParser
+            Code.ensure_loaded?(ExAws.SQS.SweetXmlParser) -> ExAws.SQS.SweetXmlParser
+            true -> raise "no XML parser found. Please add {:saxy, \"~> 1.1\"} to your mix.exs"
+          end
```

Comment:
For some reason, if I exclude this:

```elixir
config :ex_aws_sqs, parser: ExAws.SQS.SaxyParser
```

Then I always get `SweetXmlParser`, even though the `SaxyParser` condition is first, and then SweetXml always exists as a dependency because (I'm assuming) that it's an ExAws dependency somewhere, because I'm not including it in my `mix.exs`

```
11:32:38.784 [debug] Parser: ExAws.SQS.SweetXmlParser
```

Reply:
Unfortunately BroadwaySQS brings the SweetXML dependency for you. However, if you explicitly depend on Saxy, then it should be picked first, no?

Reply:
Apologies, to be clear, if you have `{:saxy, "~> 1.1"}` in your mix.exs, then it should be picked first. That's the behaviour I am seeing on ExAws.SQS on suite, which has both available. :)

Reply:
> Unfortunately BroadwaySQS brings the SweetXML dependency for you. However, if you explicitly depend on Saxy, then it should be picked first, no?

If I specify SaxyParser in the config, I get it. For some reason, if I don't, I get SweetXmlParser. I was just confused as to why because it is the first condition here.

Reply:
I see. I will investigate and see if I can reproduce it! Thanks!